### PR TITLE
Remove use of css prop

### PR DESCRIPTION
**Pull request by danieltott**
This is a fix for #469: [How to remove warning: "ui-box deprecation.."](#469)

This would allow for a pretty easy upgrade to [`ui-box` 2.0.0](https://github.com/segmentio/ui-box/releases/tag/v2.0.0), but I figured one change at a time was best.

### CODE REVIEW

**Comment by danieltott** [comment]
I don't have too much experience with glamor - I tried the `className` variation at first and it wasn't really working correctly (I'm sure I messed something up), so I went with this object spread and it worked great, plus it already handles dealing with conflicting/overriding `className` props without having to do any more work. It really seemed cleaner this way to me (as opposed to combining strings etc). But the glamor props do seem a little "magical", while with `className` it is easy to understand what is happening.

Does something like this look good?

```jsx
// Pane.js
<Box
  borderTop={_borderTop}
  borderRight={_borderRight}
  borderBottom={_borderBottom}
  borderLeft={_borderLeft}
  boxShadow={elevationStyle}
  background={theme.getBackground(background)}
  className={makeClassName(
    {
      ...css,
      ...hoverElevationStyle,
      ...activeElevationStyle
    },
    className
  )}
  {...props}
/>

// make-classname.js
import { css } from 'glamor'

export default function makeClassName(styles = {}, className) {
  const retClassName = css(styles)
  if (className) {
    return `${retClassName} ${className}`
  }
  return retClassName
}

```

**Comment by mshwery** [comment]
@danieltott the way `ui-box` was using glamor internally was via className, so there shouldn't be any issues keeping with that functionality but moving into evergreen for now. Something like this should work:

```jsx
<Box
  className={classnames(
    this.props.className,
    someOtherClassName, 
    glamorCss(cssRules).toString()
  )}
/>
```

**Comment by danieltott** [comment]
Ahh yes - I see that being used in a couple of places in Evergreen as well.

**Comment by danieltott** [comment]
@mshwery I updated the PR to use `className`

I can see here the snapshot tests failing, but I can't get them to run locally, and also they're failing on a couple of components that I didn't touch...¯\_(ツ)_/¯

**Comment by mshwery** [comment]
The snapshot failures actually might be related to using `className` instead of css, because now the order has changed.

I would recommend just updating the snapshots.

**Comment by danieltott** [comment]
OK - I'm noticing some `css-nil` class values in there, so I'm going to look into it a bit more.

**Comment by mshwery** [comment]
@danieltott I don't think the `css-nil` values are new though – that's just from the snapshot output from a set of empty styles iirc.

**Comment by danieltott** [comment]
@mshwery I guess this was the one that was actually concerning:
image

But, I can't even get tests to run locally, I'm running into messages like this for every file inside /docs when I run `yarn run test`. Still looking into it so that I can resolve the snapshot discrepancies.

```
  docs/src/components/DesignersHero.js:2:28
  ✖    2:28  Unable to resolve path to module gatsby.
```

Probably not an issue with evergreen-ui, just figured I'd note it in case it's come up before.

File: src/layers/src/Pane.js [hunk 1]
```diff
@@ -178,11 +181,17 @@ class Pane extends PureComponent {
...
-        }}
+        className={cx(
+          className,
+          glamorCss(
+            {
+              ...css,
+              ...hoverElevationStyle,
+              ...activeElevationStyle
+            },
+            className
```
Comment (mshwery) [inline src/layers/src/Pane.js]: Why include `className` here?

Comment (danieltott) [reply src/layers/src/Pane.js]: Good catch, that was left over from some experimenting I was doing at the beginning.

**Comment by mshwery** [comment]
@danieltott I think you'll need to run `yarn ava --u` to update the snapshots to get the tests to pass.

For your issue, you could `cd docs && yarn install` just to bypass that issue, but I'm not sure why you are seeing it at all.

**Pull request closed without merging**